core/src/db/engine/ExecutionEngineImpl.cpp

```diff
@@ -336,12 +336,22 @@ ExecutionEngineImpl::Search(ExecutionEngineContext& context) {
         rc.RecordSection("Scalar field filtering");
 
         // Do And
-        for (int64_t i = 0; i < entity_count_; i++) {
-            if (!list->test(i) && !bitset->test(i)) {
-                list->set(i);
+        if (!list && !bitset) {
```

Comment:
you'd better follow this style:
if (list == nullptr) {
    if (bitset == nullptr) {
    } else {
    }
} else {
    if (bitset == nullptr) {
    } else {
    }
}